File: codes/python/Code_5_06_1.py
# 小問題 t を解く関数
def shou_mondai(A, B, t):
	cl = (A + t - 1) // t  # A÷t の小数点以下切り上げ
	cr = B // t  # B÷t の小数点以下切り捨て
	if cr - cl >= 1:
		return True
	else:
		return False

A, B = map(int, input().split())
for i in range(1, B + 1):
	if shou_mondai(A, B, i) == True:
		answer = i
print(answer)

File: codes/c/Code_5_06_1.c
#include <stdio.h>
#include <stdbool.h>

// 小問題 t を解く関数
bool shou_mondai(int A, int B, int t) {
	int cl = (A + t - 1) / t; // A÷t の小数点以下切り上げ
	int cr = B / t; // B÷t の小数点以下切り捨て
	if (cr - cl >= 1) return true;
	return false;
}

int main() {
	int A, B;
	scanf("%d%d", &A, &B);
	int answer = 0;
	int i;
	for (i = 1; i <= B; i++) {
		if (shou_mondai(A, B, i) == true) {
			answer = i;
		}
	}
	printf("%d\n", answer);
	return 0;
}
